interface.go

```diff
@@ -305,17 +379,26 @@ func (c *Client) DownloadURL(u string, output io.Writer) error {
 	var redirectBuf bytes.Buffer
 
 	cs, _, err := c.agent.run(request, nil, c.chunkSize, &redirectBuf, false)
-	if err != nil {
+
+	var storageType string
+	if cs.ResponseHeader != nil {
+		storageType = cs.ResponseHeader.Get("x-taskcluster-artifact-storage-type")
```

Comment:
Can you add a sentence or two to the doc comments of this function to say that it queries the `X-Taskcluster-Artifact-Storage-Type` header, and how its value is used to determine whether it is an error artifact or not?